src/Symfony/Component/Form/Extension/Validator/Type/FormTypeValidatorExtension.php

```diff
@@ -29,10 +29,19 @@ class FormTypeValidatorExtension extends BaseValidatorExtension
 {
     private ValidatorInterface $validator;
     private ViolationMapper $violationMapper;
-    private bool $legacyErrorMessages;
 
-    public function __construct(ValidatorInterface $validator, bool $legacyErrorMessages = true, FormRendererInterface $formRenderer = null, TranslatorInterface $translator = null)
+    /**
+     * @param FormRendererInterface|null $formRenderer
+     * @param TranslatorInterface|null   $translator
+     */
+    public function __construct(ValidatorInterface $validator, /* FormRendererInterface */ $formRenderer = null, /* TranslatorInterface */ $translator = null)
     {
+        if (\is_bool($formRenderer)) {
+            trigger_deprecation('symfony/form', '6.3', 'The signature of "%s" constructor requires 3 arguments: "ValidatorInterface $validator, FormRendererInterface $formRenderer = null, TranslatorInterface $translator = null". Passing argument $legacyErrorMessages is deprecated.', __CLASS__);
+            $formRenderer = $translator;
+            $translator = 4 <= \func_num_args() ? func_get_arg(3) : null;
```

Comment:
We should check the types of `$formRenderer` and `$translator` and throw a `TypeError` if they are neither `null` or implementations of the expected interfaces (the same in `ValidatorExtension`).